torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2226,6 +2607,10 @@ def hint():
                 ):
                     continue
 
+                # Even if we end up skipping this, we add it as "output" since it will still
+                # be checked, as noted below.
+                self._add_output_guard(sympy.Eq(self.source_to_symbol[source], expr))
+
```

Comment:
Something funny is happening here. When I tried to run training for `AlbertForMaskedLM`, there was a source not in `self.source_to_symbol`. In other words, there was a source that never went through `create_symbolic_sizes_strides_storage_offset`.

```python
KeyError: TensorPropertySource(base=LocalSource(local_name='loss'), prop=<TensorProperty.STORAGE_OFFSET: 2>, idx=None)
```

Now, there was a similar source that went through that function:

```python
AttrSource(base=LocalSource(local_name='pred'), member='loss')
```

@avikchaudhuri @ezyang 
Are they related in any way? Is this expected?

Reply:
The problem was that `create_symbolic_sizes_strides_storage_offset` was never called with that source. Nonetheless, the source did get routed to `produce_guards`.

Reply:
Yukio and I discussed this in a 1:1, Yukio to investigate more.